File: instalacao-ci/application/models/Entity/Reuniao.java
package br.edu.unipampa.app.domain;

import com.fasterxml.jackson.annotation.JsonIgnore;

import javax.persistence.*;
import java.io.Serializable;
import java.time.LocalDate;
import java.time.LocalTime;
import java.util.List;

/**
 * Created by Esther Favero on 15/05/2018.
 */
@Entity(name = "reunioes")
public class Reuniao implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String descricao;
    private LocalDate data;
    private LocalTime horario;
    private Boolean estaAberta;

    @Enumerated
    private TipoDaReuniao tipoDaReuniao;

    @JsonIgnore
    @ManyToMany(cascade = CascadeType.PERSIST, mappedBy = "reunioes", fetch = FetchType.LAZY)
    private List<Membro> membros;

    @JsonIgnore
    @OneToOne(cascade = CascadeType.ALL)
    private Moderador moderador;

    @JsonIgnore
    @OneToMany(mappedBy = "id", cascade = CascadeType.ALL, fetch = FetchType.EAGER, orphanRemoval = true)
    private List<ItemDePauta> itensDePauta;

    public Long getId() {
        return id;
    }

    public List<Membro> getMembros() {
        return membros;
    }

    public void setMembros(List<Membro> membros) {
        this.membros = membros;
    }

    public Moderador getModerador() {
        return moderador;
    }

    public void setModerador(Moderador moderador) {
        if (moderador == null) {
            throw new IllegalArgumentException();
        } else {
            this.moderador = moderador;
        }
    }

    public List<ItemDePauta> getItensDePauta() {
        return itensDePauta;
    }

    public void setItensDePauta(List<ItemDePauta> itensDePauta) {
        this.itensDePauta = itensDePauta;
    }

    public String getDescricao() {
        return descricao;
    }

    public void setDescricao(String descricao) {
        if (descricao == null || descricao == "") {
            throw new IllegalArgumentException();
        } else {
            this.descricao = descricao;
        }
    }

    public LocalDate getData() {
        return data;
    }

    public void setData(LocalDate data) {
        if (data == null) {
            throw new IllegalArgumentException();
        } else {
            this.data = data;
        }
    }

    public LocalTime getHorario() {
        return horario;
    }

    public void setHorario(LocalTime horario) {
        if (horario == null) {
            throw new IllegalArgumentException();
        } else {
            this.horario = horario;
        }
    }

    public void setReuniaoOrdinaria(TipoDaReuniao tipoDaReuniao) {
        this.tipoDaReuniao = TipoDaReuniao.ORDINARIA;
    }

    public void setReuniaoExtraordinaria(TipoDaReuniao tipoDaReuniao) {
        this.tipoDaReuniao = TipoDaReuniao.EXTRAORDINARIA;
    }

    public Boolean getEstaAberta() {
        return estaAberta;
    }

    public void setEstaAberta(Boolean estaAberta) {
        this.estaAberta = estaAberta;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public TipoDaReuniao getTipoDaReuniao() {
        return tipoDaReuniao;
    }

    public void setTipoDaReuniao(TipoDaReuniao tipoDaReuniao) {
        this.tipoDaReuniao = tipoDaReuniao;
    }

    @Override
    public String toString() {
        return "Reuniao{" +
                "id=" + id +
                ", descricao='" + descricao + '\'' +
                ", data=" + data +
                ", horario=" + horario +
                ", estaAberta=" + estaAberta +
                ", moderador=" + moderador +
                '}';
    }

}

File: instalacao-ci/application/models/Entity/Reuniao.php
<?php

namespace Entity;


use Doctrine\Common\Collections\ArrayCollection;


/**
 * @Entity(repositoryClass="repositories\Reunioes_repository")
 * @Table(name="reunioes")
 */
class Reuniao
{

    /**
     * @Id
     * @Column(type="integer", nullable=false)
     * @GeneratedValue(strategy="IDENTITY")
     */
    private $id;

    /**
     * @Column(type="string", nullable=false)
     */
    private $descricao;

    /**
     * @Column(type="date", nullable=false)
     */
    private $data;

    /**
     * @Column(type="time", nullable=false)
     */
    private $horario;

    /**
     * @Column(type="boolean", nullable=false)
     */
    private $estaAberta;

    /**
     * @Column(type="string", nullable=false)
     */
    private $tipoDaReuniao;

    /**
     * @ManyToMany(targetEntity="Entity\Membro", mappedBy="reunioes")
     * @JoinTable(name="membros_reunioes")
     */
    private $membros;

    /**
     * @OneToOne(targetEntity="Entity\Moderador")
     */
    private $moderador;

    /**
     * @OneToMany(targetEntity="Entity\ItemDePauta", mappedBy="reuniao")
     */
    private $itensDePauta;

    /**
     * Reuniao constructor.
     */
    public function __construct()
    {
        $this->membros = new ArrayCollection();
        $this->itensDePauta = new ArrayCollection();
    }

    /**
     * @return mixed
     */
    public function getId()
    {
        return $this->id;
    }

    /**
     * @param mixed $id
     */
    public function setId($id)
    {
        $this->id = $id;
    }

    /**
     * @return mixed
     */
    public function getDescricao()
    {
        return $this->descricao;
    }

    /**
     * @param mixed $descricao
     */
    public function setDescricao($descricao)
    {
        $this->descricao = $descricao;
    }

    /**
     * @return mixed
     */
    public function getData()
    {
        return $this->data;
    }

    /**
     * @param mixed $data
     */
    public function setData($data)
    {
        $this->data = $data;
    }

    /**
     * @return mixed
     */
    public function getHorario()
    {
        return $this->horario;
    }

    /**
     * @param mixed $horario
     */
    public function setHorario($horario)
    {
        $this->horario = $horario;
    }

    /**
     * @return mixed
     */
    public function getEstaAberta()
    {
        return $this->estaAberta;
    }

    /**
     * @param mixed $estaAberta
     */
    public function setEstaAberta($estaAberta)
    {
        $this->estaAberta = $estaAberta;
    }

    /**
     * @return mixed
     */
    public function getTipoDaReuniao()
    {
        return $this->tipoDaReuniao;
    }

    /**
     * @param mixed $tipoDaReuniao
     */
    public function setTipoDaReuniao($tipoDaReuniao)
    {
        if (!in_array($tipoDaReuniao, TipoDaReuniao::getAvailableTypes())) {
            throw new \InvalidArgumentException("Invalid type");
        }

        $this->tipoDaReuniao = $tipoDaReuniao;
    }

    /**
     * @return mixed
     */
    public function getMembros()
    {
        return $this->membros;
    }

    /**
     * @param mixed $membros
     */
    public function setMembros($membros)
    {
        $this->membros = $membros;
    }

    /**
     * @return mixed
     */
    public function getModerador()
    {
        return $this->moderador;
    }

    /**
     * @param mixed $moderador
     */
    public function setModerador($moderador)
    {
        $this->moderador = $moderador;
    }

    /**
     * @return mixed
     */
    public function getItensDePauta()
    {
        return $this->itensDePauta;
    }

    /**
     * @param mixed $itensDePauta
     */
    public function setItensDePauta($itensDePauta)
    {
        $this->itensDePauta = $itensDePauta;
    }

}
